### PR TITLE
Fix/clean path

### PR DESCRIPTION
**Pull request by nibra**
Feature - check path length

Related to joomla/joomla-cms#32567

### CODE REVIEW

File: src/InputFilter.php [hunk 1]
```diff
@@ -1016,23 +1016,62 @@ private function cleanHtml($source)
...
+	 *
+	 * If a path matches but is longer than 4095 bytes, it is cleared.
+	 *
+	 * @param   string  $source                The path as provided; it gets cleaned in place, if possible.
+	 * @param   string  $rootPattern           The pattern to identify an absolute path (e.g., '/' on Linux, 'C:\' on Windows),
+	 * @param   string  $pathSeparatorPattern  The pattern for valid path separators
+	 * @param   string  $filePattern           The pattern for valid file and directory names
+	 * @param   string  $pathSeparator         The native path separator
+	 *
+	 * @return bool
```
Comment (richard67) [inline src/InputFilter.php:1053]:
```suggestion
	 * @return boolean
```
Fix PHPCS.

Comment (richard67) [reply src/InputFilter.php:1053]: In addition, we have many deprecated messages due to curly bracket syntax used for arrays, see https://ci.joomla.org/joomla-framework/filter/63/1/3 .

Comment (richard67) [reply src/InputFilter.php:1053]: ... But those seem not to be related to this PR, or I get the log wrong.

Comment (nibra) [reply src/InputFilter.php]: Fixed

**Comment by GeraintEdwards** [comment]
@nibra 

With respect to my issue https://github.com/joomla-framework/filter/issues/51 - all the valid paths listed in my issue pass through cleanly with this new code.  So this code works better than the current code in my view.

BUT note that  double dotted folders such as 

`/var/www/vhosts/mainsite/../../another.website.net/something`

still pass through unaffected - not sure what the intention is with the filtering as I've not seen the security bulletin

**Comment by nibra** [comment]
> BUT note that double dotted folders such as
> 
> `/var/www/vhosts/mainsite/../../another.website.net/something`
> 
> still pass through unaffected - not sure what the intention is with the filtering as I've not seen the security bulletin

That is intended. A filter should not interpret the content, in this case resolve paths. That is up to the filesystem package.

**Pull request merged**